craft_providers/lxd/launcher.py

```diff
@@ -36,6 +38,37 @@
 logger = logging.getLogger(__name__)
 
 
+class InstanceTimer(threading.Thread):
+    """Timer for update instance that still alive."""
+
+    count: int = 0
+    active: bool = True
+    instance: LXDInstance
+
+    def __init__(self, instance: LXDInstance, interval: int = 3) -> None:
+        self.instance = instance
+        self.interval = interval
+        super().__init__(daemon=True)
+
+    def run(self) -> None:
+        """Run the timer."""
+        while self.active:
+            self.count += 1
+            try:
+                self.instance.config_set("user.craft_providers.timer", str(self.count))
+                logger.debug("Set instance timer to %r", self.count)
+            except LXDError:
+                # Error in timer update is not critical
+                logging.exception("Error updating instance timer")
+            time.sleep(self.interval)
+
+        logger.debug("Instance timer update stopped.")
```

Comment:
We should probably set the config value here to "done" or another sentinel value.

Reply:
We can set it to DONE or something, but it is not used later anyway.

Reply:
I'm more thinking for user debugging if something goes wrong — if the timer is marked as done, we know it completed and something else failed. If not, we can presume the process crashed or something.